pkg/controller/podautoscaler/horizontal.go

```diff
@@ -271,15 +271,15 @@ func (a *HorizontalController) reconcileAutoscaler(hpa *autoscaling.HorizontalPo
 	rescaleReason := ""
 	timestamp := time.Now()
 
-	if currentReplicas > hpa.Spec.MaxReplicas {
+	if currentReplicas == 0 {
```

Comment:
We should use spec.replicas instead of status.replicas. Status.replicas can be (in theory, hard to tell for sure in practice) zero due to some node failures, connectivity problems, replica set hiccups and in that case we probably don't want to update the target to 0.